internal/manifests/targetallocator/targetallocator.go

```diff
@@ -0,0 +1,33 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package targetallocator
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/amazon-cloudwatch-agent-operator/internal/manifests"
+)
+
+// Build creates the manifest for the TargetAllocator resource.
+func Build(params manifests.Params) ([]client.Object, error) {
+	var resourceManifests []client.Object
+	if !params.OtelCol.Spec.TargetAllocator.Enabled {
```

Comment:
Shouldn't we also return here if prometheus is spec is empty? Since we do not need to turn TA on if prometheus is missing

Reply:
In our webhook logic, we already check to see if there's a valid prometheus configuration provided if the `target-allocator` is enabled, so adding it here may be redundant.

Reply:
Please test it and document it